crates/primitives/src/compression/mod.rs

```diff
@@ -1,9 +1,7 @@
+use alloc::vec::Vec;
 use std::{cell::RefCell, thread_local};
```

Comment:
what about this?

Reply:
yeah I found this one, seems strange that is passes the no-std compilation before no?

Reply:
yepp for sure weird

Reply:
For Refcell I think we should use `use core::cell::RefCell;` but for `thread_local`, I don't think this is available so maybe we should keep as is no?

Reply:
yepp, would be great if you can open an issue for replacing `thread_local` with smthg that works

Reply:
- Fixed RefCell
- Created the issue https://github.com/paradigmxyz/reth/issues/10871

Reply:
thanks